tests/scripts/all.sh

```diff
@@ -1268,6 +1268,175 @@ component_test_ecp_restartable_no_internal_rng () {
     # no SSL tests as they all depend on having a DRBG
 }
 
+component_test_tls1_2_default_stream_cipher_only () {
+    msg "build: default with only stream cipher"
+
+    # Disable AEAD (controlled by the presence of one of GCM_C, CCM_C, CHACHAPOLY_C
+    scripts/config.py unset MBEDTLS_GCM_C
+    scripts/config.py unset MBEDTLS_CCM_C
+    scripts/config.py unset MBEDTLS_CHACHAPOLY_C
+    # Disable CBC-legacy (controlled by MBEDTLS_CIPHER_MODE_CBC plus at least one block cipher (AES, ARIA, Camellia, DES))
+    scripts/config.py unset MBEDTLS_CIPHER_MODE_CBC
+    # Disable CBC-EtM (controlled by the same as CBC-legacy plus MBEDTLS_SSL_ENCRYPT_THEN_MAC)
+    scripts/config.py unset MBEDTLS_SSL_ENCRYPT_THEN_MAC
+    # Enable stream (currently that's just the NULL pseudo-cipher (controlled by MBEDTLS_CIPHER_NULL_CIPHER))
+    scripts/config.py set MBEDTLS_CIPHER_NULL_CIPHER
+    # Modules that depend on AEAD
+    scripts/config.py unset MBEDTLS_SSL_CONTEXT_SERIALIZATION
+    scripts/config.py unset MBEDTLS_SSL_SESSION_TICKETS
+    # Enable Key Wrapping mode for 128-bit block ciphers
```

Comment:
Thanks for providing an explanation in the commit message, but I think it would be better to have it directly in the comment here.

I don't think I agree with this change. IMO consistency with development is not a good reason, the test name says "default" so it should be aligned with the default config of the current branch, not that of another branch. If there are warnings without this, we should fix them: we're supposed to build without warnings in all configurations that pass `check_config.h`. Perhaps `MBEDTLS_SSL_TICKET_C` should also depend on `GCM || CCM || ChachaPoly`, not just `MBEDTLS_SSL_SESSION_TICKETS`?

Reply:
I also wasn't sure about this. I agree that default on this branch is a base point. So I will try to investigate this part again.

Reply:
Note that the warnings might be related to the possible confusion between `SSL_TICKET_C` and `SSL_SESSION_TICKETS` that just pointed out in the main PR. I haven't time to fully investigate this right now, but my current impression is that `SSL_TICKET_C` should depend on AEAD, hence be disabled in this build, and that would fix the warnings I think.

(I'm not sure if `SSL_SESSSION_TICKETS` should also depend on AEAD, presently I would think not.)

Reply:
I think it should be like this (`SSL_SESSSION_TICKETS` is useed in `ssl_[client/server]2.c` and it can't be used without `MBEDTLS_SSL_TICKET_C`):
- `MBEDTLS_SSL_TICKET_C` depends on `GCM || CCM || ChachaPoly`
- `SSL_SESSSION_TICKETS` depends on `MBEDTLS_SSL_TICKET_C`

Reply:
Added comment with proposition of the solution in main PR https://github.com/Mbed-TLS/mbedtls/pull/6374#discussion_r990651762.